tests/sandbox/EqualityComparable.h

```diff
@@ -0,0 +1,76 @@
+// FIXME remove pragma after moving to cts
+#pragma once
+
+#include "../common/common.h"
+
+#include "TypeTraits.h"
+
+#include <chrono>
+#include <iostream>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+template <typename It>
+constexpr bool check_equality_comparable_requirement(It valid_iterator) {
+  // static_assert(type_traits::has_equality_operator_v<It>);
+  
+  // Expect to fail a compilation if equal operator is not implemented
+  static_cast<void>((valid_iterator == valid_iterator));
```

Comment:
Not great to have compilation failure: a single failure would result in entire `cpp` file disabled from tests with all test cases and assertions within.
Why not implemet some C++20-like type trait an not have something like:
```
constexpr bool is_equally_comparable =
    sycl_cts::util::type_traits::is_equally_comparable_v<It>;
if constexpr (!is_equally_comparable) {
  STATIC_REQUIRE(is_equally_comparable);
} else {
   STATIC_CHECK(...);
   STATIC_CHECK(...);
  ...
}
```

You might even use sfinae instead of `constexpr if` if needed

Reply:
I think c++ 20 style is using concepts instead of writing from scratch check functions, but still done it with type trait using))